paddle/operators/send_op.cc

```diff
@@ -51,8 +67,12 @@ class SendOp : public framework::OperatorBase {
     detail::RPCClient* rpc_client = client_var->GetMutable<detail::RPCClient>();
 
     for (size_t i = 0; i < ins.size(); i++) {
-      VLOG(3) << "sending " << ins[i] << " to " << epmap[i];
-      rpc_client->AsyncSendVariable(epmap[i], ctx, scope, ins[i]);
+      if (IsVariableInitialized(scope, ins[i])) {
+        VLOG(3) << "sending " << ins[i] << " to " << epmap[i];
+        rpc_client->AsyncSendVariable(epmap[i], ctx, scope, ins[i]);
+      } else {
+        VLOG(3) << "don't send no-initialied variable: " << ins[i];
```

Comment:
Sorry I don't understand why it's possible that the send OP will execute when its inputs are not initialized, is it a special thing for the select row OP?

Reply:
If we split a var(selected_rows) into multiple vars(selected_rows) by [split_selected_rows](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/split_selected_rows_op.cc) operator, not all the outputs have the value, for example:

The Tnput var(X):
```
X.rows = [1,2,3]
X.height = 10
height_sections=[4, 4, 2]
```
After split_selected_rows:
The outputs vars(Out)
```bash
Out[0].rows = [1,2, 3]
Out[0].height= 4

Out[1].rows = []
Out[1].height = 4

Out[2].rows = []
OUt[2].height = 2
```

the output var which rows is empty would not be initialized, which means it hasn't called `mutable_data` interface to allocate memory.

Reply:
Thanks!